src/base/Algorithms.hh

```diff
@@ -43,6 +44,24 @@ CELER_CONSTEXPR_FUNCTION auto move(T&& v) noexcept ->
     return static_cast<typename std::remove_reference<T>::type&&>(v);
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Support swapping of trivial types.
+ */
+template<class T>
+CELER_FORCEINLINE_FUNCTION void trivial_swap(T& a, T& b) noexcept
+{
+    static_assert(std::is_trivially_copy_constructible<T>::value,
+                  "Value is not trivially copyable");
+    static_assert(std::is_trivially_move_assignable<T>::value,
+                  "Value is not trivially movable");
+    static_assert(std::is_trivially_destructible<T>::value,
+                  "Value is not trivially destructible");
+    T temp{a};
```

Comment:
Couldn't this be:
```
T temp{::celeritas::move(a)};
```
(and/or why does the other benefit from move and no that one) ?

Reply:
Yes, you're completely right... not sure how I missed that. Probably in the shuffle of redefining swaps and moves and...